service/src/test/java/de/adorsys/psd2/sandbox/xs2a/pis/PaymentInitiationWithScaSteps.java

```diff
@@ -39,38 +45,72 @@ public void initiatePayment(String paymentProduct) {
-    debtorAccount.put("currency", "EUR");
-    debtorAccount.put("iban", "DE51250400903312345678");
+    HashMap<String, String> debtorAccount =  createAccount("DE51250400903312345678");
     payment.setDebtorAccount(debtorAccount);
 
     Amount instructedAmount = new Amount();
     instructedAmount.setAmount("520");
     instructedAmount.setCurrency("EUR");
     payment.setInstructedAmount(instructedAmount);
 
-    HashMap<String, String> creditorAccount = new HashMap<>();
-    creditorAccount.put("currency", "EUR");
-    creditorAccount.put("iban", "DE15500105172295759744");
+    HashMap<String, String> creditorAccount = createAccount("DE15500105172295759744");
     payment.setCreditorAccount(creditorAccount);
 
     payment.setCreditorAgent("AAAADEBBXXX");
     payment.setCreditorName("WBG");
 
-    Address creditorAddress = new Address();
-    creditorAddress.setBuildingNumber("56");
-    creditorAddress.setCity("Nürnberg");
-    creditorAddress.setCountry("DE");
-    creditorAddress.setPostalCode("90543");
-    creditorAddress.setStreet("WBG Straße");
+    Address creditorAddress = createCreditorAddress();
+    payment.setCreditorAddress(creditorAddress);
+
+    payment.setRemittanceInformationUnstructured("Ref. Number WBG-1222");
+
+    Request<PaymentInitiationSctJson> request = new Request<>();
+    request.setBody(payment);
+    request.setHeader(headers);
+
+    ResponseEntity<PaymentInitationRequestResponse201> response = template.exchange(
+        context.getPaymentService() + "/" +
+            context.getPaymentProduct(),
+        HttpMethod.POST,
+        request.toHttpEntity(),
+        PaymentInitationRequestResponse201.class);
+
+    context.setPaymentId(response.getBody().getPaymentId());
+  }
+
+  @Given("^PSU initiated a future-dated payment using the payment product (.*)$")
+  public void initiateFutureDatedPayment(String paymentProduct) {
+    context.setPaymentService("payments");
+    context.setPaymentProduct(paymentProduct);
+
+    HashMap<String, String> headers = TestUtils.createSession();
+
+    PaymentInitiationSctJson payment = new PaymentInitiationSctJson();
+    payment.setEndToEndIdentification("WBG-123456789");
+
+    HashMap<String, String> debtorAccount =  createAccount("DE51250400903312345678");
+    payment.setDebtorAccount(debtorAccount);
+
+    Amount instructedAmount = new Amount();
+    instructedAmount.setAmount("520");
+    instructedAmount.setCurrency("EUR");
+    payment.setInstructedAmount(instructedAmount);
+
+    HashMap<String, String> creditorAccount = createAccount("DE15500105172295759744");
+    payment.setCreditorAccount(creditorAccount);
+
+    payment.setCreditorAgent("AAAADEBBXXX");
+    payment.setCreditorName("WBG");
+
+    Address creditorAddress = createCreditorAddress();
     payment.setCreditorAddress(creditorAddress);
 
     payment.setRemittanceInformationUnstructured("Ref. Number WBG-1222");
+    payment.setRequestedExecutionDate(LocalDate.of(2019,03,20));
```

Comment:
Use `LocalDate.now()` to avoid errors starting on 20th of March 2019

Reply:
done